core/src/main/java/org/bf2/srs/fleetmanager/execution/impl/workers/ScheduleRegistryWorker.java

```diff
@@ -88,17 +117,59 @@ public void execute(Task aTask, WorkerContext ctl) throws RegistryStorageConflic
         ctl.delay(() -> tasks.submit(ProvisionRegistryTenantTask.builder().registryId(registry.getId()).build()));
     }
 
+    private RegistryData createRegistry(RegistryData registryData, AccountInfo accountInfo) throws EvalInstancesNotAllowedException, AccountManagementServiceException, TooManyEvalInstancesForUserException, TermsRequiredException, ResourceLimitReachedException, RegistryStorageConflictException {
+        // Figure out if we are going to create a standard or eval instance.
+        ResourceType resourceType = evalInstancesOnlyEnabled ?
+                ResourceType.REGISTRY_INSTANCE_EVAL : accountManagementService.determineAllowedResourceType(accountInfo);
+
+        if (resourceType == ResourceType.REGISTRY_INSTANCE_EVAL) {
+            // Are eval instances allowed?
+            if (!evalInstancesEnabled) {
+                throw new EvalInstancesNotAllowedException();
+            }
+
+            // Limit the # of eval instances per user.  Need to check storage for list of eval registry instances.
+            List<RegistryData> registriesByOwner = storage.getRegistriesByOwner(accountInfo.getAccountUsername());
+            int evalInstanceCount = 0;
+            for (RegistryData ownedRegistry : registriesByOwner) { // TODO Perform a dedicated query
+                if (RegistryInstanceTypeValueDto.EVAL.value().equals(ownedRegistry.getInstanceType())) {
+                    evalInstanceCount++;
+                }
+            }
+            if (evalInstanceCount >= maxEvalInstancesPerUser) {
+                throw new TooManyEvalInstancesForUserException();
+            }
+        }
+
+        // Try to consume some quota from AMS for the appropriate resource type (standard or eval).  If successful
+        // we'll get back a subscriptionId - if not we'll throw an exception.
+        String subscriptionId = accountManagementService.createResource(accountInfo, resourceType);
```

Comment:
We should split `createResource` to separate out the terms check, which can be done in-request (I think it does not require knowing the resource type).

Reply:
I don't see it this way. In the UI case, the terms check is done before calling the API, so we do not need to check it until this point. I want to return the info to the UI as soon as possible, making as many external requests as possible. async.